packages/dev/s2-docs/src/client.tsx

```diff
@@ -15,34 +17,51 @@ let updateRoot = hydrate({
 // and in a React transition, stream in the new page. Once complete, we'll pushState to
 // update the URL in the browser.
 async function navigate(pathname: string, push = false) {
+  setNavigationLoading(true, pathname);
+  
   try {
     let res = await fetchRSC<ReactElement>(pathname.replace('.html', '.rsc'));
     let currentPath = location.pathname;
     let [newBasePath, newPathAnchor] = pathname.split('#');
 
-    updateRoot(res, () => {
-      if (push) {
-        history.pushState(null, '', pathname);
-        push = false;
-      }
+    startTransition(() => {
```

Comment:
updateRoot already calls `startTransition` internally so you might not need this?